Cardinality API: add the ability to filter by "active" series (#5136)
grafana/mimir: docs/sources/mimir/references/http-api/index.md

Returns realtime label values cardinality associated to request param `label_names[]` across all ingesters, for the authenticated tenant, in `JSON` format.
It returns the series count per label value associated to request param `label_names[]`.

> Two methods of counting are available, `inmemory` and `active`. This choice is controlled via the `count_method` parameter.

Reviewer: ```suggestion
Two methods of counting are available: `inmemory` and `active`. To choose one, use the `count_method` parameter.
```

The `inmemory` method counts the number of series in currently opened TSDBs in Mimir's ingesters.

> Two subsequent calls may return completely different results if an ingester cut a block between calls.

Reviewer: ```suggestion
Two subsequent calls might return completely different results if an ingester cut a block between calls.
```

> Two subsequent calls may return completely different results, if ingester did a block cutting between the calls.

Reviewer 2: ```suggestion
Two subsequent calls may return completely different results if an ingester cut a block between calls.
```

Is this the only case where this is true? Or is "block cutting between the calls" one example scenario (of many possibilities) which could cause this?

This method of counting is most useful for understanding ingester memory usage.

> The `active` counting method counts also uses values from currently opened TSDBs in ingesters, but it filters out series that have not received a sample within active series idle timeout.

Reviewer 2: ```suggestion
The `active` method also counts series in currently opened TSDBs in Mimir's ingesters, but filters out series that have not received a sample within a configurable duration of time. 
```

Reviewer 2: honestly I still don't understand how if `active` is just a subset of `inmemory` (you said it "filters stuff out" iiuc) how its not also impacted by this same block cutting problem? But I don't necessarily think I need to understand this =)

Reviewer 2: Thank you for explaining!

> Use the `ingester.active-series-metrics-idle-timeout` parameter to configure this duration.

Reviewer: ```suggestion
To configure this duration, use the `ingester.active-series-metrics-idle-timeout` parameter.
```

> Use the `ingester.active-series-metrics-idle-timeout` parameter to configure this duration.

Reviewer 3: ```suggestion
Use the `-ingester.active-series-metrics-idle-timeout` parameter to configure this duration.
```

> The duration of this timeout is controlled via the `ingester.active-series-metrics-idle-timeout` parameter.

Reviewer 2: ```suggestion
Use the `ingester.active-series-metrics-idle-timeout` parameter to configure this duration.
```

> This method of counting is most useful for understanding which label values are represented in the samples ingested by Mimir in the last `ingester.active-series-metrics-idle-timeout`.

Reviewer: ```suggestion
This method of counting is most useful for understanding what label values are represented in the samples ingested by Mimir in the last `ingester.active-series-metrics-idle-timeout`.
```

> This method of counting is most useful for understanding which label values are represented in the samples ingested by Mimir in the last `ingester.active-series-metrics-idle-timeout`.

Reviewer 3: ```suggestion
This method of counting is most useful for understanding which label values are represented in the samples ingested by Mimir in the last `-ingester.active-series-metrics-idle-timeout`.
```

> This method of counting is most useful for understanding which label values are contributing to the current sample ingestion in Mimir.

Reviewer 2: ```suggestion
This method of counting is most useful for understanding which label values are represented in the samples ingested by Mimir in the last `ingester.active-series-metrics-idle-timeout` minutes.
```

I'm not necessarily sure that its minutes but does this phrasing work?

Author: Yep, the unit is specified in the value. So you can say `5m` to mean 5 minutes or `30s` to mean 30 seconds.

> Two subsequent calls will likely return similar results, as this window of time is not related to the block cutting on ingesters.

Reviewer: ```suggestion
Two subsequent calls will likely return similar results, because this window of time is not related to the block cutting on ingesters.
```

> Values will change only as a result of changes in the ingestion rate of Mimir.

Reviewer 2: ```suggestion
Values will change only as a result of changes in the data ingested by Mimir. 
```

Its not necessarily the rate right? If the rate goes up because I go from 4 dpm to 16 dpm but I'm still sending the same label set, I would expect the answer to remain the same right?
